### PR TITLE
build: merge dtpool Makefile and update dependency

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description

Having dtpool target in the same Makefile allows full dependency
tracking of testing programs.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [ ] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Have read and agree to the Yaksa CLA terms (https://github.com/pmodels/yaksa/wiki/Yaksa-Contributor-License-Agreement)

### CODE REVIEW

**Comment by hzhou** [comment]
test:yaksa

**Comment by hzhou** [comment]
@pavanbalaji The Jenkins review is having hiccup (which still puzzles me). Meanwhile, could you start review?

**Comment by hzhou** [comment]
test:yaksa

File: Makefile.am [hunk 1]
```diff
@@ -25,13 +25,15 @@ libyaksa_la_LDFLAGS = -version-info @libyaksa_so_version@
 
 include $(top_srcdir)/src/Makefile.mk
 
-testlists=
+test_deps = libyaksa.la test/dtpools/libdtpools.la
+
+testlists =
```
Comment (pavanbalaji) [inline Makefile.am:30]: Unrelated white space cleanup.  Should be in a separate patch.

File: test/Makefile.mk [hunk 2]
```diff
@@ -5,11 +5,12 @@
 
 EXTRA_DIST += runtests.py
 
-SUBDIRS = test/dtpools
-
 LDADD = libyaksa.la test/dtpools/libdtpools.la
+
```
Comment (pavanbalaji) [inline test/Makefile.mk:9]: Unrelated white space cleanup.  Should be in a separate patch.

File: Makefile.am (hunk 1 above)
Comment (hzhou) [reply Makefile.am]: fixed.

File: test/Makefile.mk (hunk 2 above)
Comment (hzhou) [reply test/Makefile.mk]: fixed.

**Comment by hzhou** [comment]
test:yaksa

File: Makefile.am [hunk 3]
```diff
@@ -25,13 +25,15 @@ libyaksa_la_LDFLAGS = -version-info @libyaksa_so_version@
 
 include $(top_srcdir)/src/Makefile.mk
 
+test_deps = libyaksa.la test/dtpools/libdtpools.la
```
Comment (pavanbalaji) [inline Makefile.am:28]: Shouldn't `test/dtpools/libdtpools.la` be added to `test_deps` inside `test/dtpools/Makefile.mk`?  That's the Makefile that's generating this library.

Comment (hzhou) [reply Makefile.am:28]: Sounds good.

**Comment by hzhou** [comment]
test: yaksa

File: Makefile.am (hunk 3 above)
Comment (hzhou) [reply Makefile.am]: done.

Comment (pavanbalaji) [reply Makefile.am]: Also, `test_deps` should be a dependency for the test files themselves, not for `make testing`.  Otherwise, when someone builds the tests, the dependency won't reflect correctly.

Comment (hzhou) [reply Makefile.am]: That means we need spell out the dependency for each test program... why doesn't `automake` do that for me (since it's in `LDADD`). Let me first check see if I missed anything...

Comment (hzhou) [reply Makefile.am]: It turns out we can remove this `test_deps`. Each test program already has them as dependencies.

**Comment by hzhou** [comment]
test: yaksa

**Comment by hzhou** [comment]
test: yaksa
